doc/networking.md

```diff
@@ -0,0 +1,132 @@
+# Livepeer Networking Protocol v2 Spec
+
+## High level flow
+
+![broadcaster-transcoder network v2 1](https://user-images.githubusercontent.com/292510/41455677-c8437268-7032-11e8-9ce8-bfdd9b6e3fc0.png)
+[Sequence diagram source](https://sequencediagram.org/index.html#initialData=C4S2BsFMAIBkQG6QA6UgJ2gOUsA7gPboDWIAdgObQIBMAUHcgIbqgDGIzZw0ASpBRABnYOgCedAELoCTACZsmIjAFoAfP0EjxALmgB5dGwAWkbU2BE+kAI51Nw0WJXrpshUuAY9hk2dEWVvxCyAxu8orK6Oq+puaW6DoAOmQAKuhMZEJsBHIY1nbhHlEAPC6x-hkJOumZ2bn5AJJkAGYEDAzgBAShRZFe0Wq1WTl5idAAygIAtpDcBVIyEZ4YZSrD9WN6UxSz88Ghc3J0QA)
+
+## Broadcaster to Registry
+
+The broadcaster does an on-chain lookup to retrieve the orchestrator's ServiceURI in order to initiate an transcoding job.
+
+## Broadcaster to Orchestrator:
+
+### gRPC `GetTranscoder`
+
+This is called by the broadcaster to request information on a transcoder from an orchestrator. The request contains the following data:
+
+```protobuf
+// This request is sent by the broadcaster in `GetTranscoder` to request
+// information on which transcoder to use.
+message TranscoderRequest {
+
+  // ID of the job that the broadcaster needs a transcoder for
+  int64 jobId = 1;
+
+  // Broadcaster's signature over the jobId
+  bytes sig = 2;
+}
+```
+
+The `sig` field is the broadcaster's signature using its Ethereum key, over the jobId expressed as a string. The signature consists of the resulting bytes without any further encoding.
+`sig = broadcaster.Sign(jobId.String())`
+
+Verification of `TranscoderRequest` consists of the following steps:
+1. Fetch the job based on JobId.
+    * TODO distinguish if an orchestrator is lagging behind and hasn't yet received the block with the job. Might require incorporating the node's **current** block number into the request. `CreationBlock`  as stored in the job only approximates when the job is submitted to the chain, not when it is included in a block and assigned to a transcoder.
```

Comment:
Yeah -- in the split-brain case, the best thing to do would be to return an appropriate error code (eg, 404 "Job Not Found" or "Block Not Found"), and have the client be responsible for re-trying later. That way we don't tie up any resources on the server waiting for a block that may never come.

Also not sure if we really need the block number, or would the job ID be enough given enough time/retries?

I'm inclined to wait on this case and make it watertight after the merge, since the client doesn't yet do other corrective action as a result of server errors.

Reply:
I think returning an appropriate error if the orchestrator cannot find the job seems reasonable. And yeah I think if we're just returning an error and not trying to handle any block syncing disparities between the orchestrator and broadcaster in favor of having the broadcaster just retry , the job ID should be enough